code/modules/clothing/spacesuits/rig/suits/station.dm

```diff
@@ -158,9 +159,10 @@
 	suit_type = "hazmat hardsuit"
 	desc = "An Anomalous Material Interaction hardsuit that protects against the strangest energies the universe can throw at it."
 	icon_state = "science_rig"
-	armor = list(melee = 45, bullet = 5, laser = 45, energy = 80, bomb = 60, bio = 100, rad = 100)
+	armor = list(melee = 45, bullet = 5, laser = 40, energy = 65, bomb = 60, bio = 100, rad = 100)
 	slowdown = 1
 	offline_vision_restriction = 1
+	emp_protection = 40
```

Comment:
the cell charge reduction is the biggest part of EMP vs hardsuits. And no, that shock proc has nothing to do with it at all.

This line is the problem: if(cell) cell.emp_act(severity_class + 15)

I'd say change it to...
if(cell) cell.emp_act((severity_class + 15)*1+(0.1*emp_protection))

And perhaps increase the EMP protection values a bit